pkg/server/server.go

```diff
@@ -652,6 +664,9 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 				return
 			}
 
+			if len(recvCh) > 9 {
```

Comment:
nit: I would declare the other constant (recvCh size) so the two never get out of sync.